src/server.js

```diff
@@ -0,0 +1,136 @@
+import { Server, Model, RestSerializer } from "miragejs";
+import {
+  loginHandler,
+  signupHandler,
+} from "./backend/controllers/AuthController";
+import {
+  getHistoryVideosHandler,
+  addVideoToHistoryHandler,
+  removeVideoFromHistoryHandler,
+  clearHistoryHandler,
+} from "./backend/controllers/HistoryController";
+import {
+  getAllVideosHandler,
+  getVideoHandler,
+} from "./backend/controllers/VideoController";
+import { videos } from "./backend/db/videos";
+import { categories } from "./backend/db/categories";
+import {
+  getAllCategoriesHandler,
+  getCategoryHandler,
+} from "./backend/controllers/CategoryController";
+import {
+  getLikedVideosHandler,
+  addItemToLikedVideos,
+  removeItemFromLikedVideos,
+} from "./backend/controllers/LikeController";
+import {
+  getAllPlaylistsHandler,
+  addNewPlaylistHandler,
+  removePlaylistHandler,
+  getVideosFromPlaylistHandler,
+  addVideoToPlaylistHandler,
+  removeVideoFromPlaylistHandler,
+} from "./backend/controllers/PlaylistController";
+import { users } from "./backend/db/users";
+import {
+  addItemToWatchLaterVideos,
+  getWatchLaterVideosHandler,
+  removeItemFromWatchLaterVideos,
+} from "./backend/controllers/WatchLaterController";
+export function makeServer({ environment = "development" } = {}) {
+  return new Server({
+    serializers: {
+      application: RestSerializer,
+    },
+    environment,
+    // TODO: Use Relationships to have named relational Data
+    models: {
+      video: Model,
+      category: Model,
+      user: Model,
+      like: Model,
+      history: Model,
+      playlist: Model,
+      watchlater: Model,
+    },
+
+    // Runs on the start of the server
+    seeds(server) {
+      server.logging = false;
+      videos.forEach((item) => {
+        server.create("video", { ...item });
+      });
+      categories.forEach((item) => server.create("category", { ...item }));
+      users.forEach((item) =>
+        server.create("user", {
+          ...item,
+          likes: [],
+          watchlater: [],
+          history: [],
+          playlists: [],
+        })
+      );
+    },
+
+    routes() {
+      this.namespace = "api";
+      // auth routes (public)
+      this.post("/auth/signup", signupHandler.bind(this));
+      this.post("/auth/login", loginHandler.bind(this));
```

Comment:
why we need to bind every `route` here?

Reply:
This is of the mockbee, I haven't made any change